encoders/protoc-gen-firebase-encoders/src/main/kotlin/com/google/firebase/encoders/proto/codegen/Parsing.kt

```diff
@@ -27,9 +33,128 @@ interface DescriptorParser {
     fun parse(protoFiles: List<FileDescriptorProto>): Collection<UserDefined>
 }
 
+/**
+ * Default [DescriptorParser] implementation.
+ *
+ * The parser repackages all input messages into the `vendorPackage` of the provided [config].
+ * Additionally it only returns messages that are marked for inclusion in [config].
+ *
+ * Any extensions that are encountered are added directly to extended messages as fields, so
+ * extensions are indistinguishable from message's own fields.
+ */
 class DefaultParser @Inject constructor(private val config: CodeGenConfig) : DescriptorParser {
-    override fun parse(protoFiles: List<FileDescriptorProto>): Collection<UserDefined> {
-        return listOf()
+    override fun parse(files: List<FileDescriptorProto>): List<UserDefined> {
+        val parsedMessages = files.asSequence().flatMap { file ->
+            val javaPackage = "${config.vendorPackage}.${file.`package`}"
+
+            val parent = Owner.Package(file.`package`, javaPackage, file.name)
+            parseEnums(parent, file.enumTypeList).plus(parseMessages(parent, file.messageTypeList))
+        }.toList()
```

Comment:
do you need to convert it to list first?

Reply:
good point, removed